app/views/interactions/index.html.erb

```diff
@@ -29,12 +29,12 @@
           </td>
           <td class="name">
             <%= interaction.label %>
-            <p><%= display_link_for(interaction) %></p>
+            <p><%= link = display_link_for(interaction) %></p>
           </td>
           <td>
           </td>
           <td class='text-center'>
-            <%= link_to('Edit link', edit_local_authority_service_interaction_links_path(interaction_slug: interaction.slug), class: "btn btn-default btn-s") %>
+            <%= link_to(link == 'No link' ? 'Add link' : 'Edit link', edit_local_authority_service_interaction_links_path(interaction_slug: interaction.slug), class: "btn btn-default btn-s") %>
```

Comment:
This is slightly unusual - we've coupled the text of the action button to the text we display for the link under the label.  If we change `display_link_for` to return some more markup, or just swap from `'No link'` to `'No Link'` we're going to have to change this conditional.  It'd be better to add a new helper that does something similar to `display_link_for` by looking for a `Link` instance for the supplied `interaction` and rendering the appropriate text based on finding or not finding that `Link`.

This helper would probably be v. similar to `display_link_for` and so could point us in the direction of wanting some kind of presenter for this page to wrap up the logic and get things both out of the view and out of helpers.  But maybe we don't need to yet - it's something we might consider later as this page gets more complex.